pkg/infrabin/grpc.go

```diff
@@ -0,0 +1,44 @@
+package infrabin
+
+import (
+	"log"
+	"net"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
+)
+
+// Server wraps the gRPC server and implements infrabin.Infrabin
+type GRPCServer struct {
+	Name   string
+	Server *grpc.Server
+}
+
+// Listen binds the server to the indicated interface:port.
+func (s *GRPCServer) ListenAndServe() {
+	ln, err := net.Listen("tcp", "0.0.0.0:50051")
+	if err != nil {
+		log.Fatalf("Listen failed on 0.0.0.0:50051: %v", err)
+	}
+
+	log.Printf("Starting %s grpc server on %s", s.Name, ln.Addr())
```

Comment:
```suggestion
	log.Printf("Starting %s server on %s", s.Name, ln.Addr())
```
Just to avoid having `Starting grpc grpc server`, where the first one is coming from `s.Name`